### PR TITLE
Updated length on all global e164 numbers

### PR DESCRIPTION
**Pull request by pimjansen**
### What is the reason for this PR?

It seems that our e164 generates a lot of invalid numbers. I did a cross check on all of the formats and adjusted the length where needed. I validated it against (https://libphonenumber.appspot.com/phonenumberparser?number=%2B68227265).

Please note that i did not check the full format of the number, only the sizing

- [ ] A new feature
- [x] Fixed an issue

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [ ] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

**Comment by pimjansen** [comment]
I won't say it fixes #300 but it will help get the correct format in place. Could be that some local providers still extend the list though but for now i kept this out of the scope.

**Comment by GrahamCampbell** [comment]
Can you update the BC check config please to allow this change?

**Comment by pimjansen** [comment]
@GrahamCampbell updated, seems that we only allowed A-Z where the e164 needs 1-9 in the regex as well.

File: roave-bc-check.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 parameters:
     ignoreErrors:
-        - '#\[BC\] CHANGED: Property Faker\\Provider\\[a-zA-Z]+::\$[a-zA-Z]+ changed default value#'
-        - '#\[BC\] CHANGED: Property Faker\\Provider\\[a-z_A-Z]+\\[a-zA-Z]+::\$[a-zA-Z]+ changed default value#'
+        - '#\[BC\] CHANGED: Property Faker\\Provider\\[a-zA-Z]+::\$[1-9a-zA-Z]+ changed default value#'
```
Comment (bramceulemans) [inline roave-bc-check.yaml:3]: Why not go from 0-9 here?

Comment (pimjansen) [reply roave-bc-check.yaml:3]: There is no usecase for such class at this point so thats the makn reason.

Comment (pimjansen) [reply roave-bc-check.yaml:3]: @bramceulemans can you review again?

**Comment by bramceulemans** [comment]
Will do!

**Comment by bramceulemans** [comment]
LGTM now

**Review by bramceulemans — APPROVED** [review]

**Comment by pimjansen** [comment]
Thanks @bramceulemans

**Pull request merged**